sound/soc/soc-pcm.c

```diff
@@ -1370,7 +1375,8 @@ static bool dpcm_end_walk_at_be(struct snd_soc_dapm_widget *widget,
 
 	if (dir == SND_SOC_DAPM_DIR_OUT) {
 		for_each_card_rtds(card, rtd) {
-			if (!rtd->dai_link->no_pcm)
+			if (!rtd->dai_link->no_pcm ||
+			    !strcmp(widget->name, SOC_VIRT_DAI_PLAYBACK))
```

Comment:
a new widget flag is better rather than string checking.

Reply:
strings should be my next fight...

Reply:
@lgirdwood to this your comment: do I understand it correctly, that you're proposing a new token for this widget type? A new type would be too much? And then translate that into a new flag in the "dapm control" section of `struct snd_soc_dapm_widget`? BTW, I think `VIRTUAL_WIDGET()` isn't suitable for this, is it?